### PR TITLE
improve German translation

### PR DESCRIPTION
**Pull request by otacke**

### CODE REVIEW

**Comment by thomasmars** [comment]
Thanks, this looks good. It should be reviewed for the next release.

File: language/de.json [hunk 1]
```diff
@@ -103,28 +105,32 @@
         {
           "label": "Option hätte nicht ausgewählt werden sollen",
           "default": "Hätte nicht ausgewählt werden sollen"
+        },
+        {
+          "label": "Text für den \"Benötigt Antwort\"-Hinweis",
+          "default": "Bitte antworte erst, damit du die Lösung sehen kannst",
```
Comment (timothyylim) [inline language/de.json]: Trailing comma, invalid json.

**Comment by timothyylim** [comment]
@otacke , good work so far :)

Shameless plug: I would recommend to try my language file checker 

https://github.com/h5p/h5p-cli/pull/11

To automate the testing next time

**Comment by otacke** [comment]
Thanks ;-)

**Comment by timothyylim** [comment]
Cool! Just missing some defaults:

```
Missing default false in:
 {
  "label": "Show confirmation dialog on \"Check\"",
  "default": false
}

Missing default false in:
 {
  "label": "Show confirmation dialog on \"Retry\"",
  "default": false
}

Missing default false in:
 {
  "label": "Automatically check answers",
  "default": false,
  "description": "Enabling this option will make accessibility suffer, make sure you know what you're doing."
}
```

Then we should be good to go!

**Comment by otacke** [comment]
Will fix it in a flash ... Hell, what did I do ...

**Comment by timothyylim** [comment]
The defaults were missing from before but the language checker checks for json validation first.

**Comment by otacke** [comment]
Alright, now I am confused. Semantics has the default label for "Show confirmation dialog on \"Check\"" and "Show confirmation dialog on \"Retry\"", and "Automatically check answers" was added later.

**Comment by timothyylim** [comment]
Yes, I think it was probably because a new feature was added for automatically checking answers.

**Comment by otacke** [comment]
Wouldn't git resolve that when merging?

**Comment by otacke** [comment]
Just merged my changes with the latest version.

**Comment by timothyylim** [comment]
Missing two descriptions and a few defaults. Sorry I didn't manage to catch these earlier!

```
Missing description "Hint for the user. This will appear before user checks his answer/answers." in:
 {
  "label": "Tip text",
  "description": "Hint for the user. This will appear before user checks his answer/answers."
}

Missing description "Feedback text, variables available: @score and @total. Example: 'You got @score of @total points.'" in:
 {
  "label": "Feedback text",
  "default": "You got @score of @total points",
  "description": "Feedback text, variables available: @score and @total. Example: 'You got @score of @total points.'"
}

Missing default true in:
 {
  "label": "Enable \"Retry\" button",
  "default": true
}

Missing default true in:
 {
  "label": "Enable \"Show Solution\" button",
  "default": true
}

Missing default "auto" in:
 {
  "label": "Question Type",
  "description": "Select the look and behaviour of the question.",
  "default": "auto",
  "options": [
    {
      "label": "Automatic"
    },
    {
      "label": "Multiple Choice (Checkboxes)"
    },
    {
      "label": "Single Choice (Radio Buttons)"
    }
  ]
}

Missing default false in:
 {
  "label": "Give one point for the whole task",
  "description": "Enable to give a total of one point for multiple correct answers. This will not be an option in \"Single answer\" mode.",
  "default": false
}

Missing default true in:
 {
  "label": "Randomize answers",
  "description": "Enable to randomize the order of questions on display.",
  "default": true
}

Missing default true in:
 {
  "label": "Require answer before the solution can be viewed",
  "default": true
}

Missing default false in:
 {
  "label": "Disable image zooming for question image",
  "default": false
}

Missing default false in:
 {
  "label": "Show confirmation dialog on \"Check\"",
  "default": false
}

Missing default false in:
 {
  "label": "Show confirmation dialog on \"Retry\"",
  "default": false
}

Missing default false in:
 {
  "label": "Automatically check answers",
  "default": false,
  "description": "Enabling this option will make accessibility suffer, make sure you know what you're doing."
}
```

**Comment by otacke** [comment]
Right, I should have checked for additions to _semantics.json_ because my changes to the language file are quite dated. I'll add the missing descriptions as soon as I come around to it.

If I understand the remarks correctly, default values should also be defined within the language files? Wouldn't that mean (unnecessary) redundancy to the definitions in _semantics.json_, or is there a need to define different default values depending on the language?

**Comment by timothyylim** [comment]
Very good point. It doesn't make much sense to add defaults for booleans or numeric values but it some cases it would make sense, strings for example. 

I think if the descriptions are updated we could probably go ahead and merge :)

**Comment by otacke** [comment]
Thanks for the clarification. I quickly added the missing stuff during my lunch break directly on github, so I cannot check everything properly. Hope everythings fine, because as of this evening I'll be away until Sunday.

What the new "importance" label in _semantics.json_ by the way?

**Comment by timothyylim** [comment]
@otacke, I think if you add the two descriptions it should be good to go 👍 

Nice catch, the new importance values will be used to better index the content types and make them more understandable to a search engine :)

**Comment by otacke** [comment]
Seems that I edited my master branch the day before yesterday instead of this one ...

**Comment by timothyylim** [comment]
Cool, would you mind running it through a linter first?

jsonlint.com

I think that there may be some errors

**Comment by otacke** [comment]
It says: "Valid JSON."

**Comment by timothyylim** [comment]
Hm. Something doesn't look right here:

https://github.com/otacke/h5p-multi-choice/blob/f8fad65b9e205a7ea0673f1b7c0994afeb4dcd26/language/de.json#L224

**Comment by otacke** [comment]
That's strange, indeed. I'll check it when I am back home on Sunday when I have some "real" time for looking into it.

File: language/de.json [hunk 2]
```diff
@@ -1,228 +1,228 @@
...
+          "label": "Option hätte ausgewählt werden sollen",
+          "default": "Hätte ausgewählt werden müssen"
+        },
+        {
+          "label": "Option hätte nicht ausgewählt werden sollen",
+          "default": "Hätte nicht ausgewählt werden sollen"
+        },
+      {
+        "label": "Text für \"Benötigt Antwort\"-Hinweis",
+        "default": "Eine Antwort ist notwendig, bevor die Lösung angesehen werden kann.",
```
Comment (timothyylim) [inline language/de.json]: trailing comma

**Comment by otacke** [comment]
Not done yet.

**Comment by otacke** [comment]
There we go. Eventually, I took my time and was rewarded with "Check passed." by your language checker.

**Comment by timothyylim** [comment]
👍 

The language checker is still very much in pre-pre-alpha phase but I'm glad you gave it a go

**Comment by otacke** [comment]
I'll probably use it to contribute some more translations to German in the near future. Nice tool!

**Pull request merged**